demos/device_defender_for_aws/report_builder.c

```diff
@@ -44,53 +49,68 @@
 /* Formats used to generate the JSON report. */
 #define JSON_PORT_OBJECT_FORMAT \
     "{"                         \
-    "\"port\": %u"              \
+    "\"%s\": %u"                \
     "},"
 
-#define JSON_CONNECTION_OBJECT_FORMAT     \
-    "{"                                   \
-    "\"local_port\": %u,"                 \
-    "\"remote_addr\": \"%u.%u.%u.%u:%u\"" \
+#define JSON_CONNECTION_OBJECT_FORMAT \
+    "{"                               \
+    "\"%s\": %u,"                     \
+    "\"%s\": \"%u.%u.%u.%u:%u\""      \
     "},"
 
 #define JSON_REPORT_FORMAT_PART1 \
     "{"                          \
-    "\"header\": {"              \
-    "\"report_id\": %u,"         \
-    "\"version\": \"%u.%u\""     \
+    "\"%s\": {"                  \
+    "\"%s\": %u,"                \
+    "\"%s\": \"%u.%u\""          \
     "},"                         \
-    "\"metrics\": {"             \
-    "\"listening_tcp_ports\": {" \
-    "\"ports\": "
+    "\"%s\": {"                  \
+    "\"%s\": {"                  \
+    "\"%s\": "
 
 #define JSON_REPORT_FORMAT_PART2 \
     ","                          \
-    "\"total\": %u"              \
+    "\"%s\": %u"                 \
     "},"                         \
-    "\"listening_udp_ports\": {" \
-    "\"ports\": "
-
-#define JSON_REPORT_FORMAT_PART3     \
-    ","                              \
-    "\"total\": %u"                  \
-    "},"                             \
-    "\"network_stats\": {"           \
-    "\"bytes_in\": %u,"              \
-    "\"bytes_out\": %u,"             \
-    "\"packets_in\": %u,"            \
-    "\"packets_out\": %u"            \
-    "},"                             \
-    "\"tcp_connections\": {"         \
-    "\"established_connections\": {" \
-    "\"connections\": "
-
-#define JSON_REPORT_FORMAT_PART4 \
+    "\"%s\": {"                  \
+    "\"%s\": "
+
+#define JSON_REPORT_FORMAT_PART3 \
     ","                          \
-    "\"total\": %u"              \
-    "}"                          \
+    "\"%s\": %u"                 \
+    "},"                         \
+    "\"%s\": {"                  \
+    "\"%s\": %u,"                \
+    "\"%s\": %u,"                \
+    "\"%s\": %u,"                \
+    "\"%s\": %u"                 \
+    "},"                         \
+    "\"%s\": {"                  \
+    "\"%s\": {"                  \
+    "\"%s\": "
+
+#define JSON_REPORT_FORMAT_PART4   \
+    ","                            \
+    "\"%s\": %u"                   \
+    "}"                            \
+    "}"                            \
+    "},"                           \
+    "\"%s\": {"                    \
```

Comment:
Can the macro names for the custom metric keys be used here in the format string?

Reply:
No; they cant be used as the macro expansion has parenthesis.

Reply:
That seems like the use of parentheses is reducing the use of the macro then.

Reply:
That is accurate